### PR TITLE
M3-4630: Refactor Domain Create workflow

### PR DESCRIPTION
**Pull request by tiffwong**
## Description
As mentioned in [https://github.com/linode/manager/pull/5646](https://github.com/linode/manager/pull/5646), the Domain Create flow is inconsistent with the rest of Cloud's creation flows. To fix this, Domain Create has been moved to it's own page with a corresponding route, `/domains/create`.

## Type of Change
- Non breaking change ('update', 'change')

## Note to Reviewers
Please check the different paths that lead to the Domain Create workflow including non-CMR and CMR.

### CODE REVIEW

File: packages/manager/src/features/Domains/CreateDomain/CreateDomain.tsx [hunk 1]
```diff
@@ -0,0 +1,717 @@
...
+
+  componentDidMount() {
+    this.mounted = true;
+  }
+
+  componentWillUnmount() {
+    this.mounted = false;
+  }
+
+  componentDidUpdate(prevProps: CombinedProps, prevState: State) {
```
Comment (Jskobos) [inline packages/manager/src/features/Domains/CreateDomain/CreateDomain.tsx:212]: This pattern is bad and we should refactor it (later)

File: packages/manager/src/features/Domains/CreateDomain/CreateDomain.tsx [hunk 2]
```diff
@@ -0,0 +1,717 @@
...
+                <FormControlLabel
+                  value="slave"
+                  label="Slave"
+                  control={<Radio />}
+                  data-qa-domain-radio="Slave"
+                  disabled={disabled}
+                />
+              </RadioGroup>
+              <TextField
+                errorText={'' && errorMap.domain}
```
Comment (Jskobos) [inline packages/manager/src/features/Domains/CreateDomain/CreateDomain.tsx:331]: Need to delete the `'' &&`, this always evaluates to false

File: packages/manager/src/features/Domains/CreateDomain/CreateDomain.tsx [hunk 3]
```diff
@@ -0,0 +1,717 @@
...
+  };
+
+  updateInsertDefaultRecords = (value: DefaultRecordsType) => {
+    this.setState({ defaultRecordsSetting: value });
+  };
+
+  updateType = (
+    e: React.ChangeEvent<HTMLInputElement>,
+    value: 'master' | 'slave'
+  ) => this.setState({ type: value });
```
Comment (Jskobos) [inline packages/manager/src/features/Domains/CreateDomain/CreateDomain.tsx:666]: I think this should reset the error state too

File: packages/manager/src/features/Domains/CreateDomain/CreateDomain.tsx [hunk 4]
```diff
@@ -0,0 +1,717 @@
...
+                    If specified, we can automatically create some domain
+                    records (A/AAAA and MX) to get you started, based on one of
+                    your Linodes or NodeBalancers.
+                  </FormHelperText>
+                </React.Fragment>
+              )}
+              {isCreatingMasterDomain &&
+                this.state.defaultRecordsSetting === 'linode' && (
+                  <React.Fragment>
+                    <LinodeSelect
```
Comment (Jskobos) [inline packages/manager/src/features/Domains/CreateDomain/CreateDomain.tsx:402]: Can these be disabled? If possible they should be, for restricted users.

**Review by Jskobos — CHANGES_REQUESTED** [review]
Good work, most of my comments are about old stuff that has been wonky for ages. Only blocking change is the broken errormap check for the domain name.

**Review by WilkinsKa1 — COMMENTED** [review]
This is great, nice work on this! Besides the error missing that Jared called out, I also noticed this: 
Screen Shot 2020-11-12 at 9 49 29 AM
may just need a max-width applied or some adjustments to the flex container.

**Review by Jskobos — APPROVED** [review]
Looks good, if possible please also disable the default records dropdown and nodebalancers dropdown for restricted users before merging.

**Review by WilkinsKa1 — APPROVED** [review]

**Pull request merged**